File: pages/api/send-sms.ts
import type { NextApiRequest, NextApiResponse } from 'next';
import { TwilioConfig } from '@/utils/notification';
import twilio from 'twilio';

type ResponseData = {
  success: boolean;
  message: string;
  messageId?: string;
};

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse<ResponseData>
) {
  if (req.method !== 'POST') {
    return res.status(405).json({ success: false, message: 'Μόνο POST αιτήματα επιτρέπονται' });
  }

  try {
    const { phoneNumber, message, config } = req.body;

    if (!phoneNumber || !message) {
      return res.status(400).json({ 
        success: false, 
        message: 'Απαιτούνται ο αριθμός τηλεφώνου και το μήνυμα' 
      });
    }

    // Validate Twilio config
    if (!config || !config.accountSid || !config.authToken || !config.phoneNumber) {
      return res.status(400).json({ 
        success: false, 
        message: 'Λείπει η ρύθμιση του Twilio. Παρακαλώ προσθέστε τα στοιχεία στις ρυθμίσεις.' 
      });
    }

    // Για τώρα, επιστρέφουμε εικονική επιτυχία χωρίς να στέλνουμε πραγματικά SMS
    // Αυτό θα αντικατασταθεί με την πραγματική λειτουργία αργότερα
    console.log('[MOCK] Αποστολή SMS στο:', phoneNumber);
    console.log('[MOCK] Μήνυμα:', message);
    console.log('[MOCK] Χρησιμοποιώντας ρυθμίσεις Twilio:', config.accountSid, config.phoneNumber);
    
    // Δημιουργία εικονικού ID μηνύματος
    const mockSid = 'SM' + Math.random().toString(36).substring(2, 15);
    
    return res.status(200).json({
      success: true,
      message: 'Το μήνυμα SMS στάλθηκε με επιτυχία (προσομοίωση)',
      messageId: mockSid
    });
    
    /* ΠΡΑΓΜΑΤΙΚΟΣ ΚΩΔΙΚΑΣ ΠΟΥ ΘΑ ΧΡΗΣΙΜΟΠΟΙΗΘΕΙ ΑΡΓΟΤΕΡΑ
    try {
      // Initialize the Twilio client with the provided credentials
      const client = twilio(config.accountSid, config.authToken);
      
      // Send the SMS message
      const twilioMessage = await client.messages.create({
        body: message,
        from: config.phoneNumber,
        to: phoneNumber
      });

      return res.status(200).json({
        success: true,
        message: 'Το μήνυμα SMS στάλθηκε με επιτυχία',
        messageId: twilioMessage.sid
      });
    } catch (twilioError: any) {
      console.error('Twilio error:', twilioError);
      return res.status(400).json({
        success: false,
        message: `Σφάλμα Twilio: ${twilioError.message || 'Άγνωστο σφάλμα Twilio'}`
      });
    }
    */
  } catch (error: any) {
    console.error('Error sending SMS:', error);
    return res.status(500).json({
      success: false,
      message: `Σφάλμα κατά την αποστολή SMS: ${error.message || 'Άγνωστο σφάλμα'}`
    });
  }
}

File: utils/send_sms.py
#!/usr/bin/env python3
"""
SMS sending utility for Twilio integration
This script is used to send SMS messages through Twilio API.
"""

import os
import sys
import json
from twilio.rest import Client

def send_sms(to_number, message, config=None):
    """
    Send an SMS message using Twilio
    
    Args:
        to_number (str): The recipient's phone number
        message (str): The message content
        config (dict, optional): Twilio configuration (account_sid, auth_token, phone_number)
                              If not provided, will use environment variables.
    
    Returns:
        dict: Response with success status and message SID or error
    """
    
    # Get configuration from environment or passed config
    try:
        if config:
            account_sid = config.get('accountSid')
            auth_token = config.get('authToken') 
            from_number = config.get('phoneNumber')
        else:
            account_sid = os.environ.get('TWILIO_ACCOUNT_SID')
            auth_token = os.environ.get('TWILIO_AUTH_TOKEN')
            from_number = os.environ.get('TWILIO_PHONE_NUMBER')
        
        # Validate configuration
        if not account_sid or not auth_token or not from_number:
            return {
                'success': False, 
                'error': 'Missing Twilio credentials'
            }
            
        # Initialize Twilio client
        client = Client(account_sid, auth_token)
        
        # Send the message
        message = client.messages.create(
            body=message,
            from_=from_number,
            to=to_number
        )
        
        # Return success response
        return {
            'success': True,
            'sid': message.sid
        }
        
    except Exception as e:
        # Return error response
        return {
            'success': False,
            'error': str(e)
        }

if __name__ == "__main__":
    # When run as a script, parse arguments from command line
    if len(sys.argv) < 3:
        print(json.dumps({
            'success': False,
            'error': 'Usage: send_sms.py <to_number> <message> [config_json]'
        }))
        sys.exit(1)
    
    to_number = sys.argv[1]
    message_text = sys.argv[2]
    
    # Check for config in third argument
    config_data = None
    if len(sys.argv) > 3:
        try:
            config_data = json.loads(sys.argv[3])
        except:
            print(json.dumps({
                'success': False,
                'error': 'Invalid JSON configuration'
            }))
            sys.exit(1)
    
    # Send message and output result as JSON
    result = send_sms(to_number, message_text, config_data)
    print(json.dumps(result))